text/0000-at-least-one-feature.md

```diff
@@ -191,14 +192,89 @@ there is no point of adding such a new minimal feature because there is nothing
+
+## RFC [#3283](https://github.com/rust-lang/rfcs/pull/3283)
+
+This proposal adds a new notion of a "feature base", which aims to essential provide more `default-features = false` options rather than just one forever.
+`default-features = false` is ret-conned as the first feature base.
+
+My view is that that RFC is pretty close to being this one in disguise.
+A "feature base" is just a feature.
+The fact that there is always one "feature base" is just another way of accomplishing the "at least one feature"
+restriction.
+
+The main difference is that `default-features = false` is ret-conned as a feature to allow for existing users.
+That can be thought of from the perspective of this RFC as a "one-off" feature migration to get reverse dependencies predating the use of `at-least-one-feature = true` on to a feature so they retroactively abide by the "at least one feature" rule.
+
+## "Negative features"
+
+I take a pretty dim view of this.
+It is easy to say that one wants negative features; it is harder to actually give a proper semantics for them.
+As far as I know, additive features are the only reasonable core semantics, so we have no choice but to "pre-process" away negative features before they ruin our core model.
+
+Here is an example of such a pre-processing step.
+Recall from our preliminary theory section that feature sets are feature *downsets*.
+Whatever the user writes, we must complete it with the missing features implied by what they wrote.
+That means we have this process:
+
+> Syntactic feature set
+>
+> --- fill in missing features --->
+>
+> Semantic feature downset
+
+We can add a middle step to account for "negative features"
+
+> Syntactic feature "dual set"
+>
+> --- negate negative features: add if they were not included, remove if they were --->
+>
+> Strictly positive feature set
+>
+> --- fill in missing features --->
+>
+> Semantic feature downset
+
+But not the ramifications of this: the final "fill in missing features" step may well "add back" one of the features that was removed!
+
+I think this will be confusing to users, to say the least.
+Moreover it is crucial to understand the underlying additive feature model to become proficient with Cargo, and the more we "disguise" it with superficial negative features, the harder it will be for users to understand that model.
+The syntax *impedes* learning.
+
+I think this is an awkward middle ground between this (dirt easy) and full feature migrations.
+Only full feature migrations provide the "ideal" naming of feature so users new to a library can be blissfully ignorant of whatever features previous versions of the library supported.
+
 # Prior art
 [prior-art]: #prior-art
 
-This is a well-known problem.
-See just-rejected [#3283](https://github.com/rust-lang/rfcs/pull/3283), 
-and my previous retracted [#3146](https://github.com/rust-lang/rfcs/pull/3146).
-
-I think this is much simpler than the other two.
+Nothing direct to my knowledge.
```

Comment:
Sounded like gentoo might be worth looking into for possible prior art.

Reply:
OK I will do. Thanks.